pkg/component/gardener/apiserver/hpa.go

```diff
@@ -0,0 +1,96 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package apiserver
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/component/apiserver"
+	"github.com/gardener/gardener/pkg/utils"
+)
+
+func (g *gardenerAPIServer) horizontalPodAutoscaler() *autoscalingv2.HorizontalPodAutoscaler {
+	if g.values.Autoscaling.Mode != apiserver.AutoscalingModeVPAAndHPA {
+		return nil
+	}
+
+	return g.horizontalPodAutoscalerInVPAAndHPAMode()
+}
+
+func (g *gardenerAPIServer) horizontalPodAutoscalerInVPAAndHPAMode() *autoscalingv2.HorizontalPodAutoscaler {
```

Comment:
```suggestion
```

Reply:
I would rather vote keep it as it is for similarity with the existing handlings:
https://github.com/gardener/gardener/blob/2eb806d95df0c7977b976149af4bb3057031de53/pkg/component/kubernetes/apiserver/horizontalpodautoscaler.go#L38-L50

and others.